### PR TITLE
Add `events-path` flag to send logs to unix socket domain

### PR DESCRIPTION
**Pull request by kajes**
Adds support for sending logs to a unix domain socket using a new `events-path` flag.

Example socket output when preparing an upgrade:
```json
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:57+02:00","COMMAND":"sdpctl appliance upgrade prepare","SDPCTL_VERSION":"0.0.0-dev","message":""}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","error":"version check disabled","message":"version check result"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"primary controller"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","current_version":"6.2.0+33912","message":"upgrade information","skipping":"","target_version":"6.2.0+33912","upgrading":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1, gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","file":"logserver-6.2.0-33912-release.zip","message":"checking file status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","file":"logserver-6.2.0-33912-release.zip","message":"got file status","status":""}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:10:58+02:00","message":"downloading image layers for logserver-6.2.0-33912-release.zip"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"143db7d084219658b1d5dc81071b28069505f56c5edd84ce9783d7d1b0689f38","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"d697bb59c3704518d967145e256d92921f010263f981c8471b246ecf31675d9b","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"143db7d084219658b1d5dc81071b28069505f56c5edd84ce9783d7d1b0689f38","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"83e8fb0fd284604f7bfb252c363cfece27aa2d5d764a8321a37bc871284942db","message":"downloading image layer"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:00+02:00","layer":"4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:02+02:00","layer":"143db7d084219658b1d5dc81071b28069505f56c5edd84ce9783d7d1b0689f38","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:04+02:00","layer":"143db7d084219658b1d5dc81071b28069505f56c5edd84ce9783d7d1b0689f38","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:11+02:00","layer":"83e8fb0fd284604f7bfb252c363cfece27aa2d5d764a8321a37bc871284942db","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:28+02:00","layer":"d697bb59c3704518d967145e256d92921f010263f981c8471b246ecf31675d9b","message":"download finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:11:47+02:00","file":"logserver-6.2.0-33912-release.zip2312233098","message":"uploading file"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:12:16+02:00","file":"logserver-6.2.0-33912-release.zip2312233098","message":"file upload finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:12:16+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"checking file status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:12:16+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"got file status","status":""}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:12:16+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"uploading file"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"file upload finished"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","file":"/home/larskajes/Downloads/appgate-6.2.0-33912-release.img.zip","message":"Uploaded file"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"checking file status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"got file status","status":"Ready"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"Status Ready"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","message":"Remote file path for the Controller controller://envy-10-97-208-2.devops/appgate-6.2.0-33912-release.img.zip"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for the appliance status","want":["healthy","warning","error","n/a","offline"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:29+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for the appliance status","want":["healthy","warning","error","n/a","offline"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached the wanted appliance status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached the wanted appliance status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","details":"","message":"prepare image change","result":"success","status":"completed"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for upgrade status","want":["verifying","ready"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","details":"","message":"prepare image change","result":"success","status":"completed"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:30+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for upgrade status","want":["verifying","ready"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:53+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached wanted status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:53+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for upgrade status","want":["ready","success"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:56+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached wanted status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:13:56+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Polling for upgrade status","want":["ready","success"]}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:24+02:00","appliance":"controller-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached wanted status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","appliance":"gateway-b465fbfc-379b-4569-af6c-8901105fe1e8-site1","message":"Reached wanted status"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","message":"3. Delete upgrade image appgate-6.2.0-33912-release.img.zip from Controller"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"Deleting file from repository"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","file":"appgate-6.2.0-33912-release.img.zip","message":"file deleted"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","message":"File appgate-6.2.0-33912-release.img.zip deleted from the Controller"}
{"@key":"64625177-d9cb-4d68-98be-0c3132f0adf6","@level":"info","@time":"2023-05-22T10:14:28+02:00","message":"prepare complete"}
```

Fixes: ZTP-1754

### CODE REVIEW

File: pkg/util/log.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	for k, v := range h.fields {
+		entry.Data[k] = v
+	}
+	bytes, err := h.formatter.Format(entry)
+	if err != nil {
+		return err
+	}
+
+	w, err := net.Dial(h.protocol, h.address)
```
Comment (danijeel) [inline pkg/util/log.go:54]: I don't think its a good idea to do dial each time in Fire, should be enough to do it once in the NewHook method, simliar of how the syslog logrus hook.
https://github.com/sirupsen/logrus/blob/352781de903c9dc639752a3ac08148132746e180/hooks/syslog/syslog.go#L23-L25

Comment (kajes) [reply pkg/util/log.go]: Ok, should be fixed now

**Comment by kajes** [comment]
> Could you show an example of the kind of output this sends on the socket.

Added example in PR description

File: pkg/util/log.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+var (
+	fieldMap = logrus.FieldMap{
+		logrus.FieldKeyTime:  "@time",
+		logrus.FieldKeyLevel: "@level",
+		logrus.FieldKeyMsg:   "message",
+	}
+	fields = logrus.Fields{
+		"@key": uuid.New(), // This will be unique for every command being run
```
Comment (danijeel) [inline pkg/util/log.go:26]: why is this here? Just curious. I'm out of touch of this feature and don't understand why you want this.

Comment (kajes) [reply pkg/util/log.go:26]: It's a convenience to make it easier to follow what happens at each executed command rather than trying to work it out ourselves, because the logs easily get pretty garbled together otherwise. It's not on request, but this is a pretty standard feature when dealing with event based systems to make it easier to follow.

**Comment by thomascellerier** [comment]
Nice :). I think it would be good to have a common `event_type` field that all records send here set. Similar to audit logs in SDP.
For example all appliance upgrade status updates could have `"event_type": "appliance_upgrade_status"` etc..
And then for each event type we would have a well defined set of fields.

**Comment by kajes** [comment]
> Nice :). I think it would be good to have a common `event_type` field that all records send here set. Similar to audit logs in SDP. For example all appliance upgrade status updates could have `"event_type": "appliance_upgrade_status"` etc.. And then for each event type we would have a well defined set of fields.

So should I add the `event_type` field now before merging or make it separate task? We need to settle on what event types we have and which fields should be included with them.

**Comment by thomascellerier** [comment]
I think its fine to merge and iterate on this in follow up pull requests.
Let's not announce this in any release notes until we have settled on something stable.

**Review by thomascellerier — APPROVED** [review]

**Comment by thomascellerier** [comment]
For the event_type, you can take a go at assigning something that makes sense to you and we can iterate from there.
But in general we want to be able to separate different kinds of events like:
- appliance upgrade status changes (do we include maintenance mode in here or as a separate event type?)
- container bundling
- file upload / delete
- backup progress

etc...
